{% extends "app/layout.html" %}

{% load staticfiles %}

{% block head-scripts %}
{% endblock %}
{% block content %}

<center><h1><b>HMMER Tutorial</b><h1></center>
<h2>HMMER: biosequence analysis using profile hidden Markov models</h2>
<p>HMMER is used for searching sequence databases for homologs of protein sequences, and for making protein sequence alignments. It implements methods using probabilistic models called profile hidden Markov models (profile HMMs).<p>
<p>Compared to BLAST, FASTA, and other sequence alignment and database search tools based on older scoring methodology, HMMER aims to be significantly more accurate and more able to detect remote homologs because of the strength of its underlying mathematical models.<p>
<p>Take a look at the <a href="http://eddylab.org/software/hmmer/Userguide.pdf">HMMER manual</a> for more detailed information.
<br>
<br>
<p>In this tutorial, we explain some of the features of the HMMER web application.<p>
<p>To begin, navigate to the HMMER web application via the menu bar, or go directly to <a href="http://hmmer.org/">http://hmmer.org/</a> <p>
<p>Under the 'Organism menu', select one or several organisms that you would like to search against. Then, select which database should be searched. Unlike BLAST, available databases only include protein sequences predicted from the genome.<p>
<img src={% static 'hmmer/images/hmmer_p1.jpg' %} style="border:1px groove;width:600px;height:600px">
<p><br>
<p><br>

<p>Paste your query sequence into the 'Query Sequence' field, or load it from disk. <p>
<p>Program phmmer only accepts fasta files as query input, so it's not available when then input file is not in fasta format. However, the program hmmsearch accepts several different formats of multiple sequence alignment(which also include fasta). The format will be checked after submitting the query.<p>
<p>In this example, the query is in fasta format.  Phmmer and Hmmsearch are both available.<p>
<img src={% static 'hmmer/images/hmmer_p2.jpg' %} style="border:1px groove;width:600px;height:600px">
<p><br>
<p><br>

<p>In this example, the query is in CLUSTAL MSA format, and phmmer is greyed out (not available).<p>
<img src={% static 'hmmer/images/hmmer_p3.jpg' %} style="border:1px groove;width:600px;height:600px">
<p><br>
<p><br>

<p>Once the correct HMMER program is selected, you can customize the parameters for your search. Hit the 'Search' button when finished.<p>
<img src={% static 'hmmer/images/hmmer_p4.jpg' %} style="border:1px groove;width:600px;height:600px">
<p><br>
<p><br>

<p>A page will display indicating that your query is running. You can store the indicated query link if you wish to come back to your HMMER results at a later time. Results are stored for one week.<p>
<img src={% static 'hmmer/images/hmmer_p5.jpg' %} style="border:1px groove">
<p><br>
<p><br>

<p>When the HMMER program has finished, your results will be displayed.<p>
<p>In 'Download' you can download the input file, the HMMER result (showed in 'Report Details') and the submission details including parameters used in this job. <p>
<p>In 'Report Details', reports are separated into independent sections by protein datasets. Use the drop-down list to navigate among them.<p>
<img src={% static 'hmmer/images/hmmer_p6.jpg' %} style="border:1px groove;width:600px;height:600px">
<p><br>
<p><br>
<!--
<p>If you have a user account for the I5K workspace, you can review your jobs by clicking 'Queries' tab.<p> 
<img src={% static 'hmmer/images/hmmer_p7.jpg' %} style="border:1px groove;width:800px;height:400px">
<p><br>
-->
{% endblock %}
