{% extends "app/layout.html" %}
{% load staticfiles %}
{% load pipeline %}
{% load jsonify %}

{% block head-scripts %}
  {% if iframe == True %}
    <style>
      nav {
          display: none;
      }
      body {
          padding-top: 0px;
      }
    </style>
  {% endif %}
  {% stylesheet 'blast-css' %}
  <link rel="stylesheet" type="text/css" href="//fonts.googleapis.com/css?family=Droid+Serif">
  <script>
    var dataset_list = {{ blastdb_list|safe }};
    var blastdb_type_counts = {{ blastdb_type_counts|jsonify }};
  </script>
{% endblock %}

{% block scripts %}
  {% javascript 'blast-js' %}
{% endblock %}

{% block content %}

<div class="tripal_details">

<form method="post" id="MainBlastForm" name="MainBlastForm" enctype="multipart/form-data">
{% csrf_token %}
<div class="tripal_organism-info-box tripal-info-box">
<!--<a href="docs/blast_databases.html">-->
<fieldset id="fieldset-datasets">
    <legend>BLAST Databases <a href="https://i5k.nal.usda.gov/content/blast-tutorial" target="_blank"><span id="tutorial"><span class="glyphicon glyphicon-book"></span> Tutorial</span></a></legend>
	<fieldset class="organism" id="fieldset-organism">
		<legend id="legend-Organisms">Organisms</legend>
		<div id="box-organism">
			<label><div organism="all" class="all organism-div organism-active-background"><input type="checkbox" organism="all" class="all-organism-checkbox">All organisms</div></label>
		</div>
	</fieldset>
	<div id="box-datasets">
		<div class="datasets-div all">
			<div class="dataset-title">All organisms</div>
            {% if blastdb_type_counts.genome_assembly or blastdb_type_counts.transcript %}
			<fieldset class="nucleotide">
				<legend>Nucleotide</legend>
				{% if blastdb_type_counts.genome_assembly %}<div><label><input type="checkbox" name="dataset-checkbox[]" organism="all" dataset-type="genome-assembly" class="all-dataset-checkbox nucleotide genome-assembly">Genome Assembly - {{ blastdb_type_counts.genome_assembly|default:0 }} databases</label></div>{% endif %}
				{% if blastdb_type_counts.transcript %}<div><label><input type="checkbox" name="dataset-checkbox[]" organism="all" dataset-type="transcript" class="all-dataset-checkbox nucleotide transcript">Transcript - {{ blastdb_type_counts.transcript|default:0 }} databases</label></div>{% endif %}
			</fieldset>
            {% endif %}
            {% if blastdb_type_counts.protein %}
			<fieldset class="peptide">
				<legend>Peptide</legend>
				<div><label><input type="checkbox" name="dataset-checkbox[]" organism="all" dataset-type="protein" class="all-dataset-checkbox peptide protein">Protein - {{ blastdb_type_counts.protein|default:0 }} databases</label></div>
			</fieldset>
            {% endif %}
		</div>
	</div>
</fieldset>

<!--
Enter here your input data as
<select name = "INPUT_TYPE">
    <option> Sequence in FASTA format
    <option> Accession or GI
</select>
-->

<fieldset id="fieldset-query">
	<legend id="legend-sequence">Query Sequence</legend>
	<div class="enter-query-text">Enter sequence below in <a target="_blank" href="http://en.wikipedia.org/wiki/FASTA_format">FASTA</a> format:</div>
	<textarea id="query-textarea" name="query-sequence" rows=6 cols=60 spellcheck="false" autocomplete="off"></textarea>
  <div><span id="load-examples"> (load from example: <span class="load-nucleotide txt">nucleotide</span> or <span class="load-peptide txt">peptide</span>) </span></div>
  Or load it from disk:
	<input type="file" name="query-file" class="query-file">
    <!--
	<div>
		Set subsequence: From <input type="text" name="QUERY_FROM" value="" size="10"> To <input type="text" name="QUERY_TO" value="" size="10">
	</div>
    -->
</fieldset>

<fieldset id="fieldset-program">
	<legend>Program</legend>
	<label class="program blastn">
		<input type="radio" name="program" value="blastn" checked="checked" class="program blastn">
		blastn
	</label>
	<label class="program tblastn">
		<input type="radio" name="program" value="tblastn" class="program tblastn">
		tblastn
	</label>
	<label class="program tblastx">
		<input type="radio" name="program" value="tblastx" class="program tblastx">
		tblastx
	</label>
	<label class="program blastp">
		<input type="radio" name="program" value="blastp" class="program blastp">
		blastp
	</label>
	<label class="program blastx">
		<input type="radio" name="program" value="blastx" class="program blastx">
		blastx
	</label>
	<div id="blastProgramDescription">blastn - Nucleotide vs. Nucleotide</div>
	<div id="buttons">
		<input type="button" value="Reset" class="btn_reset">
		<input type="button" value="Search" class="blast_submit">
	</div>
</fieldset>

<!-- BLAST options start -->
<div id="blast_options">
    <fieldset id="fieldset-options-blast">
        <legend></legend>
	<!-- BLASTN -->
		<div class="parms blastn-parms">
        <fieldset id="fieldset-options-blast-general-param">
        <legend>General Parameters</legend>
        <table id="general-param-table">
        <tr>
        <td>
            <span class="span-title" title="Number of aligned sequences to keep.">Max target sequences </span>
        </td>
        <td>
          <select id="max_target_seqs" name="max_target_seqs">
            <option value="10">10</option>
            <option value="50">50</option>
            <option value="100" selected>100</option>
            <option value="250">250</option>
            <option value="500">500</option>
          </select>
        </td></tr>
        <tr>
          <td>
            <span class="span-title" title="Expect value (E) for saving hits.  (Default: 10.0)">Expect threshold </span>
          </td>
          <td>
          <input type="text" id="evalue" name="evalue" value="10.0" size="10" class="input-fields">
          </td>
        </tr>
        <tr>
          <td>
            <span class="span-title" id="span_word_size" title="Length of initial exact match. (Default: 11)">Word size </span>
          </td>
          <td>
            <input type="text" id="word_size" name="word_size" value="11" size="10" class="input-fields">
          </td>
        </tr>
        </table>
        </fieldset>

        <fieldset id="fieldset-options-blast-scoring-param">
        <legend>Scoring Parameters </legend>
		<table id="blastn-scoring-param">
		<tr><td><span title="Reward for a nucleotide match.  (Default: 2)">Match score </span></td>
			<td><input type="text" id="reward" name="reward" value="2" size="10" class="input-fields"></td></tr>
			<tr><td><span title="Penalty for a nucleotide mismatch.  (Default: -3)">Mismatch score </span></td><td>
			<input type="text" id="penalty" name="penalty" value="-3" size="10" class="input-fields"></td></tr>
			<tr><td><span title="Cost to open a gap.  (Default: 5)">Gap opening penalty </span></td><td><input type="text" id="gapopen" name="gapopen" value="5" size="10" class="input-fields"></td></tr>
			<tr><td><span title="Cost to extend a gap.  (Default: 2)">Gap extension penalty </span></td><td><input type="text" id="gapextend" name="gapextend" value="2" size="10" class="input-fields"></td></tr>
			<tr><td><span title="Query strand(s) to search against database/subject.  (Default: both)">Strand </span></td>
			<td><select id="strand" name="strand">
				<option value="both" selected>both</option>
				<option value="plus">plus</option>
				<option value="minus">minus</option>
				</select>
			</td></tr>
		</table>

        </fieldset>

        <fieldset id="fieldset-options-blast-filter-param">
        <legend>Filters and Masking</legend>
        <input type="checkbox" class="chk_low_complexity" id="blastn_chk_low_complexity" value="yes" name="chk_low_complexity" checked="checked"><span title="Mask off segments of the query sequence that have low compositional complexity, as determined by the SEG program of Wootton & Federhen (Computers and Chemistry, 1993) or, for BLASTN, by the DUST program of Tatusov and Lipman (in preparation). Filtering can eliminate statistically significant but biologically uninteresting reports from the blast output (e.g., hits against common acidic-, basic- or proline-rich regions), leaving the more biologically interesting regions of the query sequence available for specific matching against database sequences. (Default: yes)"> Low complexity</span><br/>
        <input type="checkbox" class="chk_soft_masking" id="blastn_chk_soft_masking" value="true" name="chk_soft_masking" checked="checked"><span title="Apply filtering locations as soft masks. (i.e., only for finding initial matches) (Default: true)"> Mask for lookup table only</span><br/>
        </fieldset>
		</div>
	<!-- End BLASTN -->

	<!-- TBLASTN -->
		<div class="parms tblastn-parms">
        <fieldset id="fieldset-options-blast-general-param">
        <legend>General Parameters</legend>
        <table id="general-param-table">
        <tr>
        <td>
            <span class="span-title" title="Number of aligned sequences to keep.">Max target sequences </span>
        </td>
        <td>
          <select id="max_target_seqs" name="max_target_seqs">
            <option value="10">10</option>
            <option value="50">50</option>
            <option value="100" selected>100</option>
            <option value="250">250</option>
            <option value="500">500</option>
          </select>
        </td></tr>
        <tr>
          <td>
            <span class="span-title" title="Expect value (E) for saving hits.  (Default: 10.0)">Expect threshold </span>
          </td>
          <td>
          <input type="text" id="evalue" name="evalue" value="10.0" size="10" class="input-fields">
          </td>
        </tr>
        <tr>
          <td>
            <span class="span-title" id="span_word_size" title="Length of initial exact match. (Default: 6)">Word size </span>
          </td>
          <td>
            <input type="text" id="word_size" name="word_size" value="6" size="10" class="input-fields">
          </td>
        </tr>
        </table>
        </fieldset>

        <fieldset id="fieldset-options-blast-scoring-param">
        <legend>Scoring Parameters </legend>
		<table id="tblastn-scoring-param">
			<tr><td><span title="Scoring matrix name.  (Default: BLOSUM62)">Matrix </span></td>
			<td><select id="matrix" name="matrix">
				<option value="PAM30">PAM30</option>
				<option value="PAM70">PAM70</option>
				<option value="PAM250">PAM250</option>
				<option value="BLOSUM90">BLOSUM90</option>
				<option value="BLOSUM80">BLOSUM80</option>
				<option value="BLOSUM62" selected>BLOSUM62</option>
				<option value="BLOSUM50">BLOSUM50</option>
				<option value="BLOSUM45">BLOSUM45</option>
			</select></td>
			</tr>
			<tr><td><span title="Minimum score to add a word to the BLAST lookup table.  (Default: 13)">Threshold </span></td><td><input type="text" id="threshold" name="threshold" value="13" size="10" class="input-fields"></td></tr>
			<tr><td><span title="Cost to open a gap.  (Default: 11)">Gap opening penalty </span></td><td><input type="text" id="gapopen" name="gapopen" value="11" size="10" class="input-fields"></td></tr>
			<tr><td><span title="Cost to extend a gap.  (Default: 1)">Gap extension penalty </span></td><td><input type="text" id="gapextend" name="gapextend" value="1" size="10" class="input-fields">
			</td></tr>
		</table>

        </fieldset>

        <fieldset id="fieldset-options-blast-filter-param">
        <legend>Filters and Masking</legend>
        <input type="checkbox" class="chk_low_complexity" id="tblastn_chk_low_complexity" value="yes" name="chk_low_complexity" checked="checked"><span title="Mask off segments of the query sequence that have low compositional complexity, as determined by the SEG program of Wootton & Federhen (Computers and Chemistry, 1993) or, for BLASTN, by the DUST program of Tatusov and Lipman (in preparation). Filtering can eliminate statistically significant but biologically uninteresting reports from the blast output (e.g., hits against common acidic-, basic- or proline-rich regions), leaving the more biologically interesting regions of the query sequence available for specific matching against database sequences. (Default: yes)"> Low complexity</span><br/>
        <input type="checkbox" class="chk_soft_masking" id="tblastn_chk_soft_masking" value="false" name="chk_soft_masking"><span title="Apply filtering locations as soft masks. (i.e., only for finding initial matches) (Default: false)"> Mask for lookup table only</span><br/>
        </fieldset>
		</div>
	<!-- End TBLASTN -->

	<!-- TBLASTX -->
		<div class="parms tblastx-parms">
        <fieldset id="fieldset-options-blast-general-param">
        <legend>General Parameters</legend>
        <table id="general-param-table">
        <tr>
        <td>
            <span class="span-title" title="Number of aligned sequences to keep.">Max target sequences </span>
        </td>
        <td>
          <select id="max_target_seqs" name="max_target_seqs">
            <option value="10">10</option>
            <option value="50">50</option>
            <option value="100" selected>100</option>
            <option value="250">250</option>
            <option value="500">500</option>
          </select>
        </td></tr>
        <tr>
          <td>
            <span class="span-title" title="Expect value (E) for saving hits.  (Default: 10.0)">Expect threshold </span>
          </td>
          <td>
          <input type="text" id="evalue" name="evalue" value="10.0" size="10" class="input-fields">
          </td>
        </tr>
        <tr>
          <td>
            <span class="span-title" id="span_word_size" title="Length of initial exact match. (Default: 3)">Word size </span>
          </td>
          <td>
            <input type="text" id="word_size" name="word_size" value="3" size="10" class="input-fields">
          </td>
        </tr>
        </table>
        </fieldset>

        <fieldset id="fieldset-options-blast-scoring-param">
        <legend>Scoring Parameters </legend>
		<table id="tblastx-scoring-param">
			<tr><td><span title="Scoring matrix name.  (Default: BLOSUM62)">Matrix </span></td>
			<td><select id="matrix" name="matrix">
				<option value="PAM30">PAM30</option>
				<option value="PAM70">PAM70</option>
				<option value="PAM250">PAM250</option>
				<option value="BLOSUM90">BLOSUM90</option>
				<option value="BLOSUM80">BLOSUM80</option>
				<option value="BLOSUM62" selected>BLOSUM62</option>
				<option value="BLOSUM50">BLOSUM50</option>
				<option value="BLOSUM45">BLOSUM45</option>
			</select></td>
			</tr>
			<tr><td><span title="Minimum score to add a word to the BLAST lookup table.  (Default: 13)">Threshold&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></td><td><input type="text" id="threshold" name="threshold" value="13" size="10" class="input-fields"></td></tr>
			<tr><td><span title="Query strand(s) to search against database/subject.  (Default: both)">Strand </span></td><td>
				<select id="strand" name="strand">
				<option value="both" selected>both</option>
				<option value="plus">plus</option>
				<option value="minus">minus</option>
			</select></td>
			</tr>
		</table>

        </fieldset>

        <fieldset id="fieldset-options-blast-filter-param">
        <legend>Filters and Masking</legend>
        <input type="checkbox" class="chk_low_complexity" id="tblastx_chk_low_complexity" value="yes" name="chk_low_complexity" checked="checked"><span title="Mask off segments of the query sequence that have low compositional complexity, as determined by the SEG program of Wootton & Federhen (Computers and Chemistry, 1993) or, for BLASTN, by the DUST program of Tatusov and Lipman (in preparation). Filtering can eliminate statistically significant but biologically uninteresting reports from the blast output (e.g., hits against common acidic-, basic- or proline-rich regions), leaving the more biologically interesting regions of the query sequence available for specific matching against database sequences. (Default: yes)"> Low complexity</span><br/>
        <input type="checkbox" class="chk_soft_masking" id="tblastx_chk_soft_masking" value="false" name="chk_soft_masking"><span title="Apply filtering locations as soft masks. (i.e., only for finding initial matches) (Default: false)"> Mask for lookup table only</span><br/>
        </fieldset>
		</div>
	<!-- End TBLASTX -->

	<!-- BLASTP -->
		<div class="parms blastp-parms">
        <fieldset id="fieldset-options-blast-general-param">
        <legend>General Parameters</legend>
        <table id="general-param-table">
        <tr>
        <td>
            <span class="span-title" title="Number of aligned sequences to keep.">Max target sequences </span>
        </td>
        <td>
          <select id="max_target_seqs" name="max_target_seqs">
            <option value="10">10</option>
            <option value="50">50</option>
            <option value="100" selected>100</option>
            <option value="250">250</option>
            <option value="500">500</option>
          </select>
        </td></tr>
        <tr>
          <td>
            <span class="span-title" title="Expect value (E) for saving hits.  (Default: 10.0)">Expect threshold </span>
          </td>
          <td>
          <input type="text" id="evalue" name="evalue" value="10.0" size="10" class="input-fields">
          </td>
        </tr>
        <tr>
          <td>
            <span class="span-title" id="span_word_size" title="Length of initial exact match. (Default: 6)">Word size </span>
          </td>
          <td>
            <input type="text" id="word_size" name="word_size" value="6" size="10" class="input-fields">
          </td>
        </tr>
        </table>
        </fieldset>

        <fieldset id="fieldset-options-blast-scoring-param">
        <legend>Scoring Parameters </legend>
		<table id="blastp-scoring-param">
			<tr><td><span title="Scoring matrix name.  (Default: BLOSUM62)">Matrix </span></td>
			<td><select id="matrix" name="matrix">
				<option value="PAM30">PAM30</option>
				<option value="PAM70">PAM70</option>
				<option value="PAM250">PAM250</option>
				<option value="BLOSUM90">BLOSUM90</option>
				<option value="BLOSUM80">BLOSUM80</option>
				<option value="BLOSUM62" selected>BLOSUM62</option>
				<option value="BLOSUM50">BLOSUM50</option>
				<option value="BLOSUM45">BLOSUM45</option>
			</select></td>
			</tr>
			<tr><td><span title="Minimum score to add a word to the BLAST lookup table.  (Default: 11)">Threshold </span></td><td><input type="text" id="threshold" name="threshold" value="11" size="10" class="input-fields"></td></tr>
			<tr><td><span title="Cost to open a gap.  (Default: 11)">Gap opening penalty </span></td><td><input type="text" id="gapopen" name="gapopen" value="11" size="10" class="input-fields"></td></tr>
			<tr><td><span title="Cost to extend a gap.  (Default: 1)">Gap extension penalty </span></td><td><input type="text" id="gapextend" name="gapextend" value="1" size="10" class="input-fields"></td></tr>
		</table>

        </fieldset>

        <fieldset id="fieldset-options-blast-filter-param">
        <legend>Filters and Masking</legend>
        <input type="checkbox" class="chk_low_complexity" id="blastp_chk_low_complexity" value="no" name="chk_low_complexity"><span title="Mask off segments of the query sequence that have low compositional complexity, as determined by the SEG program of Wootton & Federhen (Computers and Chemistry, 1993) or, for BLASTN, by the DUST program of Tatusov and Lipman (in preparation). Filtering can eliminate statistically significant but biologically uninteresting reports from the blast output (e.g., hits against common acidic-, basic- or proline-rich regions), leaving the more biologically interesting regions of the query sequence available for specific matching against database sequences. (Default: no)"> Low complexity</span><br/>
        <input type="checkbox" class="chk_soft_masking" id="blastp_chk_soft_masking" value="false" name="chk_soft_masking"><span title="Apply filtering locations as soft masks. (i.e., only for finding initial matches) (Default: false)"> Mask for lookup table only</span><br/>
        </fieldset>
		</div>
	<!-- End BLASTP -->

	<!-- BLASTX -->
		<div class="parms blastx-parms">
        <fieldset id="fieldset-options-blast-general-param">
        <legend>General Parameters</legend>
        <table id="general-param-table">
        <tr>
        <td>
            <span class="span-title" title="Number of aligned sequences to keep.">Max target sequences </span>
        </td>
        <td>
          <select id="max_target_seqs" name="max_target_seqs">
            <option value="10">10</option>
            <option value="50">50</option>
            <option value="100" selected>100</option>
            <option value="250">250</option>
            <option value="500">500</option>
          </select>
        </td></tr>
        <tr>
          <td>
            <span class="span-title" title="Expect value (E) for saving hits.  (Default: 10.0)">Expect threshold </span>
          </td>
          <td>
          <input type="text" id="evalue" name="evalue" value="10.0" size="10" class="input-fields">
          </td>
        </tr>
        <tr>
          <td>
            <span class="span-title" id="span_word_size" title="Length of initial exact match. (Default: 6)">Word size </span>
          </td>
          <td>
            <input type="text" id="word_size" name="word_size" value="6" size="10" class="input-fields">
          </td>
        </tr>
        </table>
        </fieldset>

        <fieldset id="fieldset-options-blast-scoring-param">
        <legend>Scoring Parameters </legend>
		<table id="blastx-scoring-param">
			<tr><td><span title="Scoring matrix name.  (Default: BLOSUM62)">Matrix </span></td>
			<td><select id="matrix" name="matrix">
			<option value="PAM30">PAM30</option>
			<option value="PAM70">PAM70</option>
			<option value="PAM250">PAM250</option>
			<option value="BLOSUM90">BLOSUM90</option>
			<option value="BLOSUM80">BLOSUM80</option>
			<option value="BLOSUM62" selected>BLOSUM62</option>
			<option value="BLOSUM50">BLOSUM50</option>
			<option value="BLOSUM45">BLOSUM45</option>
			</select></td>
			</tr>
			<tr><td><span title="Minimum score to add a word to the BLAST lookup table.  (Default: 12)">Threshold </span></td><td><input type="text" id="threshold" name="threshold" value="12" size="10" class="input-fields"></td></tr>
			<tr><td><span title="Query strand(s) to search against database/subject.  (Default: both)">Strand </span></td><td>
			<select id="strand" name="strand">
			<option value="both" selected>both</option>
			<option value="plus">plus</option>
			<option value="minus">minus</option>
			</select></td>
			</tr>
			<tr><td><span title="Cost to open a gap.  (Default: 11)">Gap opening penalty </span></td><td><input type="text" id="gapopen" name="gapopen" value="11" size="10" class="input-fields"></td></tr>
			<tr><td><span title="Cost to extend a gap.  (Default: 1)">Gap extension penalty </span></td><td><input type="text" id="gapextend" name="gapextend" value="1" size="10" class="input-fields"></td></tr>
		</table>

        </fieldset>

        <fieldset id="fieldset-options-blast-filter-param">
        <legend>Filters and Masking</legend>
        <input type="checkbox" class="chk_low_complexity" id="blastx_chk_low_complexity" value="yes" name="chk_low_complexity"><span title="Mask off segments of the query sequence that have low compositional complexity, as determined by the SEG program of Wootton & Federhen (Computers and Chemistry, 1993) or, for BLASTN, by the DUST program of Tatusov and Lipman (in preparation). Filtering can eliminate statistically significant but biologically uninteresting reports from the blast output (e.g., hits against common acidic-, basic- or proline-rich regions), leaving the more biologically interesting regions of the query sequence available for specific matching against database sequences. (Default: yes)"> Low complexity</span><br/>
        <input type="checkbox" class="chk_soft_masking" id="blastx_chk_soft_masking" value="false" name="chk_soft_masking"><span title="Apply filtering locations as soft masks. (i.e., only for finding initial matches) (Default: false)"> Mask for lookup table only</span><br/>
        </fieldset>
		</div>
	<!-- End BLASTX -->
	<input type="hidden" id="low_complexity_hidden" name="low_complexity" value="yes">
	<input type="hidden" id="soft_masking_hidden" name="soft_masking" value="true">
    </fieldset>
</div>
<input type="hidden" id="click_submit_hidden" name="click_submit_hidden" value="false">
<!-- BLAST options end -->

	<div>
		<input type="button" value="Reset" class="btn_reset">
		<input type="button" value="Search" class="blast_submit">
	</div>

</div>
</form>
</div>

{% endblock %}
